repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,180 @@
+        return res
+
+    def _action_repair_confirm(self):
+        """Confirm repair and cascade confirmation to other group members."""
+        res = super()._action_repair_confirm()
+        if not self.env.context.get("skip_group_confirm"):
+            for repair in self.filtered("group_id"):
+                siblings = repair.group_id.repair_ids.filtered(
+                    lambda r, rid=repair.id: r.state == "draft" and r.id != rid
+                )
+                if siblings:
+                    siblings.with_context(
+                        skip_group_confirm=True
+                    )._action_repair_confirm()
+        return res
+
+    def action_repair_cancel(self):
+        """Cancel repair and cascade cancellation to other group members."""
+        res = super().action_repair_cancel()
+        if not self.env.context.get("skip_group_cancel"):
+            for repair in self.filtered("group_id"):
+                siblings = repair.group_id.repair_ids.filtered(
+                    lambda r, rid=repair.id: r.state not in ("cancel", "done")
+                    and r.id != rid
+                )
+                if siblings:
+                    siblings.with_context(skip_group_cancel=True).action_repair_cancel()
+        return res
+
+    def action_create_sale_order(self):
+        """Create a single Sale Order for all repairs in the same group."""
+        self.ensure_one()
+
+        if self.sale_order_id:
+            raise UserError(
+                _("Repair order %(name)s is already linked to sale order %(sale)s.")
+                % {"name": self.name, "sale": self.sale_order_id.name}
+            )
+        if not self.partner_id:
+            raise UserError(
+                _("You need to define a customer before creating a quotation.")
+            )
+
+        if not self.group_id:
+            return super().action_create_sale_order()
+
+        # Prevent duplicate sale orders within the same group
+        if self.group_id.repair_ids.filtered("sale_order_id"):
+            raise UserError(_("A sale order already exists for this group."))
+
+        repairs = self.group_id.repair_ids.filtered(
+            lambda r: not r.sale_order_id and r.state != "cancel"
+        )
+        if not repairs:
+            raise UserError(_("No valid repairs found in this group."))
+
+        # Create shared sale order with warehouse if available
+        order_vals = {
+            "partner_id": self.partner_id.id,
+            "company_id": self.company_id.id,
+        }
+        # Optionally set warehouse from the repair's picking type
+        if repairs and repairs[0].picking_type_id.warehouse_id:
+            order_vals["warehouse_id"] = repairs[0].picking_type_id.warehouse_id.id
+
+        sale_order = self.env["sale.order"].create(order_vals)
+
+        # Link all repairs and add order lines with explicit UOM
+        for repair in repairs:
+            repair.sale_order_id = sale_order
```

Comment:
It will result in x-times of create. If you are assigning the same value to all records you must do it at once:
```
repairs.write({"sale_order_id": sale_order.id})
```